src/epoch.rs

```diff
@@ -263,27 +272,109 @@ impl Epoch {
         None
     }
 
-    /// Makes a copy of self and sets the duration and time scale appropriately given the new duration
+    /// Creates an epoch from given duration expressed in given timescale.
+    /// In case of ET, TDB Timescales, a duration since J2000 is expected.
     #[must_use]
-    pub fn from_duration(new_duration: Duration, time_scale: TimeScale) -> Self {
-        match time_scale {
-            TimeScale::TAI => Self::from_tai_duration(new_duration),
-            TimeScale::TT => Self::from_tt_duration(new_duration),
-            TimeScale::ET => Self::from_et_duration(new_duration),
-            TimeScale::TDB => Self::from_tdb_duration(new_duration),
-            TimeScale::UTC => Self::from_utc_duration(new_duration),
-            TimeScale::GPST => Self::from_gpst_duration(new_duration),
-            TimeScale::QZSST => Self::from_qzsst_duration(new_duration),
-            TimeScale::GST => Self::from_gst_duration(new_duration),
-            TimeScale::BDT => Self::from_bdt_duration(new_duration),
+    pub fn from_duration(duration: Duration, ts: TimeScale) -> Self {
+        Self {
+            duration,
+            time_scale: ts,
         }
     }
 
+    pub fn to_duration_since_j1900(&self) -> Duration {
+        self.to_time_scale(TimeScale::TAI).duration
+    }
+
+    #[must_use]
+    /// Converts self to another time scale
+    pub fn to_time_scale(&self, ts: TimeScale) -> Self {
```

Comment:
```suggestion
    pub fn into_time_scale(&self, ts: TimeScale) -> Self {
```

I think that `into...` is a better name